docs/antrea-network-policy.md

```diff
@@ -1433,6 +1481,110 @@ The following kubectl commands can be used to retrieve CG resources:
+to refer to Groups in Antrea NetworkPolicy objects instead of specifying pods and
+namespace selectors every time.
+
+### Group CRD
+
+Below are some example Group specs:
+
+```yaml
+# Group that selects all Pods labeled role: db
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-sel
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+---
+# Group that selects all Pods labeled role: db in Namespace labeled env: prod.
+# This Group cannot be used by Antrea NetworkPolicy appliedTo because of the namespaceSelector.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-with-namespace
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+  namespaceSelector:
+    matchLabels:
+      env: prod
+---
+# Group that selects ipBlocks 10.0.10.0/24.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-ip-block
+spec:
+  # IPBlocks cannot be set along with PodSelector, NamespaceSelector or serviceReference.
+  ipBlocks:
+    - cidr: 10.0.10.0/24
+---
+# Group that selects serviceReference test-service in default Namespace.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-svc-ref
+spec:
+  # ServiceReference cannot be set along with PodSelector, NamespaceSelector or ipBlocks.
+  serviceReference:
+    name: test-service
+    namespace: default
+---
+# Group that has the previous Groups as childGroups.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-nested
+spec:
+  childGroups: [test-grp-sel, test-grp-ip-blocks, test-grp-svc-ref]
+```
+
+### Restrictions and Key differences from ClusterGroup
+
+Group has a similar spec with ClusterGroup. However, there are key differences and
+restrictions.
+
+- A Group can be set in an Antrea NetworkPolicy's `appliedTo` and `to`/`from` peers.
+  When set in Antrea NetworkPolicy `appliedTo` field, it cannot include
+  `namespaceSelector`, since Antrea NetworkPolicy is Namespace scoped. For example,
+  the `test-grp-with-namespace` Group in the [sample](#group-crd) cannot be
+  used by Antrea NetworkPolicy `appliedTo`.
+- Antrea will not validate referenced Group resources; it will however report an
+  error in the Antrea NetworkPolicy status if the Group does not meet the
+  `appliedTo` convention.
+  - If the convention is violated in the Antrea NetworkPolicy's `appliedTo` section
+    or for all the rules' `appliedTo` in the Antrea NetworkPolicy, then the Antrea
+    NetworkPolicy status will be updated to `Unrealizable`.
+  - If the convention is violated for part of, but not all, Antrea NetworkPolicy
+    rules, the Antrea NetworkPolicy status will be updated to `PartiallyRealizable`.
+- The `childGroup` only accepts strings, and they will be considered as names of
+  the Groups and looked for in the policy's own namespace. For example, if
+  childGroup child-0 exists in ns-2, it should not be added as part of
```

Comment:
Suggest to add `` around names: `child-0`, `ns-2`, `ns-1/parentGroup-0`, `ns-1/child-0`